### PR TITLE
Add QuickControls2 dependency to Meson and Qmake, and enable ahead-of-time compilation of QMLs

### PR DESCRIPTION
**Pull request by ntonnaett**
#884 happened because of the missing QuickControls2 dependency. I'll open an issue downstream.

Edit: Also turns on ahead-of-time compilation of QMLs for the qmake build.
https://doc.qt.io/qt-6/qtquick-deployment.html#ahead-of-time-compilation

### CODE REVIEW

**Comment by ntonnaett** [comment]
Turning on ahead-of-time compilation revealed that we use [required properties](https://doc.qt.io/qt-5/qtqml-syntax-objectattributes.html#required-properties) that were introduced with Qt 5.15. So our Qt 5.12 builds fail now.
@dyfer What do you think?

**Comment by dyfer** [comment]
I think it would be nice to support Qt 5.12 (e.g. to allow building on Ubuntu 20.04 without installing Qt from PPA). But if nobody else things it's needed, we can move the static build to Qt 5.15 (not sure which bugfix version? Ubuntu 22.04 ships with 5.15.3, so maybe not later than that?)

**Comment by dyfer** [comment]
I've pushed Qt update to https://github.com/jacktrip/jacktrip/tree/upgrade-qt-for-static-linux-build
That [build](https://github.com/dyfer/jacktrip/actions/runs/4003838305) doesn't run on Ubuntu 20.04 also doesn't seem to run on Fedora 34. I've danced around the "qt platform plugin" problem before, I don't have the energy to look into that again... but I do want to provide an up-to-date build of JT that's compatible with multiple OSs. Not sure what the best way forward is.

**Comment by ntonnaett** [comment]
@dhing1024 Could we avoid to use the required property feature? I saw you used that [here](https://github.com/jacktrip/jacktrip/blob/4eed6979347a82284adc111cedfd89162b13efec/src/gui/Settings.qml#L293).

**Comment by dhing1024** [comment]
> @dhing1024 Could we avoid to use the required property feature? I saw you used that [here](https://github.com/jacktrip/jacktrip/blob/4eed6979347a82284adc111cedfd89162b13efec/src/gui/Settings.qml#L293).

That sounds good

**Comment by ntonnaett** [comment]
> > @dhing1024 Could we avoid to use the required property feature? I saw you used that [here](https://github.com/jacktrip/jacktrip/blob/4eed6979347a82284adc111cedfd89162b13efec/src/gui/Settings.qml#L293).
> 
> That sounds good

Do you have any idea how to do that? I couldn't find anything equivalent for 5.12.

**Comment by dhing1024** [comment]
I wouldn't know, I'm still a bit new to Qt and not super familiar with nuanced differences between 5.12 and 5.15. But going back I'm actually not sure what the `required` property is doing for us currently.

**Comment by dyfer** [comment]
Actually... I'd like to correct my statement: it seems that it is actually possible to build Qt 5.15 static that _does run on Ubuntu 20.04_. We were missing some packages at build time to build the XCB plugin. @ntonnaett could you run the [static build from here](https://github.com/dyfer/jacktrip/actions/runs/4008494290) and see if you see any issues (that we didn't see before)?

If this works for you @ntonnaett, I suggest that this PR is reabased on https://github.com/jacktrip/jacktrip/tree/upgrade-qt-for-static-linux-build and this might be easier to fix than trying to re-introduce 5.12 compatibility.

**Comment by ntonnaett** [comment]
This is incredible news. I've run it on my computer. Seems fine but I only see subscribed Studios. There are some messages that are new to me. And I don't like the OpenSSL version. I would prefer if we used the latest bugfix release. Is the latest open source version 5.15.8?

```
SSL Library: OpenSSL 1.1.1q  FIPS 5 Jul 2022
Log file location: "/home/nils/.local/share/jacktrip/JackTrip/log.txt"
BDB1539 Build signature doesn't match environment
Cannot open DB environment: BDB0091 DB_VERSION_MISMATCH: Database environment version mismatch
```

**Comment by dhing1024** [comment]
> Seems fine but I only see subscribed Studios.

We took out the ability to start a Virtual Studio in public mode about 2 weeks ago, and finished the migration yesterday evening by updating the database entries on the JackTrip Labs end. They've all been migrated to private studios.

**Comment by dyfer** [comment]
> Is the latest open source version 5.15.8?

[Yes](https://download.qt.io/archive/qt/5.15/).
I went with 5.15.3 since this is the last version shipped with Ubuntu 22.04, but maybe compatibility wasn't broken in a later bugfix release. If you update the version and push to this branch I can test the build with newer Qt on Ubuntu 20.04 to see if it still works (we already know that 5.15.3 works).

**Comment by ntonnaett** [comment]
> > Is the latest open source version 5.15.8?
> 
> [Yes](https://download.qt.io/archive/qt/5.15/). I went with 5.15.3 since this is the last version shipped with Ubuntu 22.04, but maybe compatibility wasn't broken in a later bugfix release. If you update the version and push to this branch I can test the build with newer Qt on Ubuntu 20.04 to see if it still works (we already know that 5.15.3 works).

Let's do this in a separate PR. We should update the qt version also for macOS and Windows. I rebased this branch now.

**Comment by dyfer** [comment]
@ntonnaett looks like Linux shared lib builds need to be updated to Ubuntu 22.04, since 20.04 ships with Qt 5.12.

File: src/JackTrip.cpp [hunk 1]
```diff
@@ -1289,14 +1289,7 @@ int JackTrip::clientPingToServerStart()
     // ----------------------------------------------
     connect(&mTcpClient, &QTcpSocket::readyRead, this, &JackTrip::receivedDataTCP);
     connect(&mTcpClient, &QTcpSocket::connected, this, &JackTrip::receivedConnectionTCP);
-    // Enable CI builds on Ubuntu 20.04 with Qt 5.12.8
-#ifdef __linux__
```
Comment (dyfer) [inline src/JackTrip.cpp:1293]: This `ifdef` could potentially be changed into one based on a Qt version, but I guess now we are indeed going to draw a hard line for Qt 5.15, right?

Comment (ntonnaett) [reply src/JackTrip.cpp:1293]: I thought about that after I did the change. But I mean, the vs gui is broken with 5.12.

Comment (dyfer) [reply src/JackTrip.cpp:1293]: VS yes, but "novs" version could still be built on an older system with a version-based conditional.

Comment (ntonnaett) [reply src/JackTrip.cpp:1293]: True. I'll change that.

**Review by dyfer — APPROVED** [review]
Thanks! This is great. And in the process we got a working Linux static build again.

**Comment by ntonnaett** [comment]
Thanks for your review! I'm very happy with what just came from checking why a downstream package missed a dependency.

**Comment by dyfer** [comment]
Just a note: this PR also effectively fixed Linux static build, which was broken for a while. Since it's not part of the PR title, it is not mentioned in the release notes for 1.7.1-beta1. But since nobody complained, then maybe nobody needed it... (?)

**Pull request merged**